src/Reference/HasOneSql.php

```diff
@@ -64,6 +64,20 @@ private function _addField(string $fieldName, bool $theirFieldIsTitle, ?string $
         return $fieldExpression;
     }
 
+    private function getLinkNameWithoutReferenceSuffix(): string
+    {
+        $ourModel = $this->getOurModel();
+
+        return preg_replace('~_(' . preg_quote($ourModel->idField, '~') . '|id)$~', '', $this->link);
```

Comment:
Introduced in https://github.com/atk4/data/commit/e49806f9db29d9a06e011469c81e45b65408afba

I belive we should strip "our field id" or "their reference name".